apps/web/lib/slots.ts

```diff
@@ -15,26 +15,34 @@ export type GetSlots = {
   frequency: number;
   workingHours: WorkingHours[];
   minimumBookingNotice: number;
+  eventLength: number;
 };
 export type WorkingHoursTimeFrame = { startTime: number; endTime: number };
 
 const splitAvailableTime = (
   startTimeMinutes: number,
   endTimeMinutes: number,
-  frequency: number
+  frequency: number,
+  eventLength: number
 ): Array<WorkingHoursTimeFrame> => {
   let initialTime = startTimeMinutes;
   const finalizationTime = endTimeMinutes;
   const result = [] as Array<WorkingHoursTimeFrame>;
   while (initialTime < finalizationTime) {
     const periodTime = initialTime + frequency;
-    result.push({ startTime: initialTime, endTime: periodTime });
+    const slotEndTime = initialTime + eventLength;
+    /*
+    check if the slot end time surpasses availability end time of the user 
+    1 minute is added to round up the hour mark so that end of the slot is considered in the check instead of x9
+    eg: if finalization time is 11:59, slotEndTime is 12:00, we ideally want the slot to be available
+    */
+    if (slotEndTime <= finalizationTime + 1) result.push({ startTime: initialTime, endTime: periodTime });
```

Comment:
We need to add `+1`  to make sure a slot isn't skipped because it was from `11:00`-`12:00` for example and the finalization time was 11:59 (basically end of the hour/day). This fetches the right/expected number of slots.